bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd

```diff
@@ -0,0 +1,127 @@
+end
+
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/ccd_sch_029_1819_w_1a_091019.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00", encoding: 'ISO-8859-1:UTF-8'}) do |row|
+      {
+        school_id:          row['NCESSCH'].to_i.to_s,
+        school_year:        SURVEY_YEAR,
+        grades_offered_lo:  row['GSLO'],
+        grades_offered_hi:  row['GSHI'],
+        grade_pk_offered:   row['G_PK_OFFERED'] == 'Yes',
+        grade_kg_offered:   row['G_KG_OFFERED'] == 'Yes',
+        grade_01_offered:   row['G_1_OFFERED'] == 'Yes',
+        grade_02_offered:   row['G_2_OFFERED'] == 'Yes',
+        grade_03_offered:   row['G_3_OFFERED'] == 'Yes',
+        grade_04_offered:   row['G_4_OFFERED'] == 'Yes',
+        grade_05_offered:   row['G_5_OFFERED'] == 'Yes',
+        grade_06_offered:   row['G_6_OFFERED'] == 'Yes',
+        grade_07_offered:   row['G_7_OFFERED'] == 'Yes',
+        grade_08_offered:   row['G_8_OFFERED'] == 'Yes',
+        grade_09_offered:   row['G_9_OFFERED'] == 'Yes',
+        grade_10_offered:   row['G_10_OFFERED'] == 'Yes',
+        grade_11_offered:   row['G_11_OFFERED'] == 'Yes',
+        grade_12_offered:   row['G_12_OFFERED'] == 'Yes',
+        grade_13_offered:   row['G_13_OFFERED'] == 'Yes',
+      }
+    end
+  end
+end
+
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/ccd_sch_129_1819_w_1a_091019.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00", encoding: 'ISO-8859-1:UTF-8'}) do |row|
+      {
+        school_id:      row['NCESSCH'].to_i.to_s,
+        school_year:    SURVEY_YEAR,
+        virtual_status: VIRTUAL_SCHOOL_MAP[row['VIRTUAL_TEXT']],
+        title_i_status: TITLE_I_MAP[row['TITLEI_STATUS_TEXT']]
+      }
+    end
+  end
+end
+
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/EDGE_GEOCODE_PUBLICSCH_1819.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00", encoding: 'ISO-8859-1:UTF-8'}) do |row|
+      {
+        school_id:      row['NCESSCH'].to_i.to_s,
+        school_year:    SURVEY_YEAR,
+        community_type: COMMUNITY_TYPE_MAP[row['LOCALE']]
+      }
+    end
+  end
+end
+
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/School_lvl_demo.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00"}) do |row|
+      row = row.to_h.map {|k, v| [k, sanitize_string_for_db(v)]}.to_h
+
+      {
+        school_id:      row['School ID - NCES Assigned [Public School] Latest available year'].to_i.to_s,
+        school_year:    SURVEY_YEAR,
+
+        students_total:     row['Total Students All Grades (Includes AE) [Public School] 2018-19'].presence.try {|v| v.to_i == 0 ? nil : v.to_i},
```

Comment:
Why do we need the extra logic here?

Reply:
Schools presumably shouldn't have 0 total students -- not sure this is definitely the right approach, but is at least consistent with the prior import.

Reply:
Ok, I'm not sure if I'm reading this right or not, but it seems like this line just turns a 0 to a nil.  This is ok, but then I'm curious why we didn't do this for all of the other values.  At any rate, leaving as-is to be consistent with prior imports sounds good.

Reply:
Yeah I think you're reading it right -- I think the thought it that it's plausible for a school to have, say, 0 Hawaiian/Pacific Islander students, but not plausible to have 0 total students. So we wipe total students but not sub-groups.